### PR TITLE
Globally replace the string "dataflash" where it now means "logger"

### PR DESCRIPTION
**Pull request by magicrub**
Globally replace the string "dataflash" where it now means "logger". This is almostly purely a non-functional change.

- first few commits in vehicle code are all just comment changes
- next few library commits are get_singleton() calls with a renamed variable
- next is SITL which renamed "dataflash.bin" to "logger.bin". This is the only functional change
- next renamed a few defines within AP_Logger that apply to the term logger. AP_Logger_Dataflash still exists and is not changed.

### CODE REVIEW

**Comment by peterbarker** [comment]
dataflash.bin is actually better; it is pretending to be a dataflash device, or at least implementing the same interface for testing purposes.

File: ArduPlane/parachute.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ void Plane::parachute_release()
     if (parachute.release_in_progress()) {
         return;
     }
-    // send message to gcs and dataflash
+    // send message to gcs and logger
```
Comment (peterbarker) [inline ArduPlane/parachute.cpp]: It's not actually doing this.

File: ArduSub/fence.cpp [hunk 2]
```diff
@@ -41,7 +41,7 @@ void Sub::fence_check()
             //            }
         }
 
-        // log an error in the dataflash
+        // log an error
```
Comment (peterbarker) [inline ArduSub/fence.cpp]: Seriously, just remove this sort of comment.  There are several like it and they add absolutely nothing.

**Comment by magicrub** [comment]
potato, potato.

File: libraries/AP_Logger/AP_Logger.cpp [hunk 3]
```diff
@@ -159,8 +159,8 @@ void AP_Logger::Init(const struct LogStructure *structures, uint8_t num_types)
 #endif
 
 #ifdef HAL_LOGGING_DATAFLASH
-    if (_params.backend_types & DATAFLASH_BACKEND_BLOCK) {
-        if (_next_backend == DATAFLASH_MAX_BACKENDS) {
+    if (_params.backend_types &LOGGER_BACKEND_BLOCK) {
```
Comment (peterbarker) [inline libraries/AP_Logger/AP_Logger.cpp]: whitespace issue

File: libraries/AP_Logger/AP_Logger_SITL.cpp [hunk 4]
```diff
@@ -26,14 +26,14 @@ extern const AP_HAL::HAL& hal;
 void AP_Logger_SITL::Init()
 {
     if (flash_fd == 0) {
-        flash_fd = open("dataflash.bin", O_RDWR|O_CLOEXEC, 0777);
+        flash_fd = open(AP_Logger_SITL::filename, O_RDWR|O_CLOEXEC, 0777);
         if (flash_fd == -1) {
-            flash_fd = open("dataflash.bin", O_RDWR | O_CREAT | O_CLOEXEC, 0777);
+            flash_fd = open(AP_Logger_SITL::filename, O_RDWR | O_CREAT | O_CLOEXEC, 0777);
```
Comment (peterbarker) [inline libraries/AP_Logger/AP_Logger_SITL.cpp]: Don't need to namespace filename here.

File: libraries/AP_Logger/examples/AP_Logger_AllTypes/AP_Logger_AllTypes.cpp [hunk 5]
```diff
@@ -91,19 +91,19 @@ class AP_LoggerTest_AllTypes : public AP_HAL::HAL::Callbacks {
...
 };
 
-void AP_LoggerTest_AllTypes::flush_dataflash(AP_Logger &_dataflash)
+void AP_LoggerTest_AllTypes::flush_logger(AP_Logger &_logger)
 {
 #if CONFIG_HAL_BOARD == HAL_BOARD_SITL || CONFIG_HAL_BOARD == HAL_BOARD_LINUX
-    _dataflash.flush();
+    _logger.flush();
 #else
     // flush is not available on e.g. px4 as it would be a somewhat
```
Comment (peterbarker) [inline libraries/AP_Logger/examples/AP_Logger_AllTypes/AP_Logger_AllTypes.cpp]: What's px4?

File: libraries/AP_NavEKF2/AP_NavEKF2.cpp [hunk 6]
```diff
@@ -611,9 +611,9 @@ bool NavEKF2::InitialiseFilter(void)
     _framesPerPrediction = uint8_t((EKF_TARGET_DT / (_frameTimeUsec * 1.0e-6) + 0.5));
 
     // see if we will be doing logging
-    AP_Logger *dataflash = AP_Logger::get_singleton();
-    if (dataflash != nullptr) {
-        logging.enabled = dataflash->log_replay();
+    AP_Logger *logger = AP_Logger::get_singleton();
```
Comment (peterbarker) [inline libraries/AP_NavEKF2/AP_NavEKF2.cpp]: No `get_singleton` calls should remain on AP_Logger!  It should all be `AP::logger()`

**Comment by peterbarker** [comment]
Most of the modified comments should just be removed where they state exactly what the next line does without adding anything about the implications of doing that.  They just add noise - and work when someone decides they want to rename a whole subsystem!

File: ArduPlane/parachute.cpp (hunk 1 above)
Comment (magicrub) [reply ArduPlane/parachute.cpp]: removed comment

File: libraries/AP_Logger/AP_Logger_SITL.cpp (hunk 4 above)
Comment (magicrub) [reply libraries/AP_Logger/AP_Logger_SITL.cpp]: can't check for nulls with a reference, they need to stay for now. We can update it on another PR.

**Comment by magicrub** [comment]
fixed up and ready to merge after CI

File: libraries/AP_Logger/AP_Logger_SITL.cpp (hunk 4 above)
Comment (magicrub) [reply libraries/AP_Logger/AP_Logger_SITL.cpp]: fixed

File: ArduSub/fence.cpp (hunk 2 above)
Comment (magicrub) [reply ArduSub/fence.cpp]: removed a bunch of them

File: libraries/AP_Logger/AP_Logger.cpp (hunk 3 above)
Comment (magicrub) [reply libraries/AP_Logger/AP_Logger.cpp]: I don't see it now, the rebase must have fixed it

File: libraries/AP_Logger/examples/AP_Logger_AllTypes/AP_Logger_AllTypes.cpp (hunk 5 above)
Comment (magicrub) [reply libraries/AP_Logger/examples/AP_Logger_AllTypes/AP_Logger_AllTypes.cpp]: poof!

File: libraries/AP_NavEKF2/AP_NavEKF2.cpp (hunk 6 above)
Comment (magicrub) [reply libraries/AP_NavEKF2/AP_NavEKF2.cpp]: needs to stay for now because we can't null check a reference very well

File: ArduCopter/motors.cpp [hunk 7]
```diff
@@ -217,7 +217,7 @@ bool Copter::init_arm_motors(const AP_Arming::Method method, const bool do_armin
     // finally actually arm the motors
     motors->armed(true);
 
-    // log arming to dataflash
+    // log arming
```
Comment (peterbarker) [inline ArduCopter/motors.cpp]: Missed one.

File: ArduCopter/motors.cpp [hunk 8]
```diff
@@ -276,7 +276,7 @@ void Copter::init_disarm_motors()
     set_land_complete(true);
     set_land_complete_maybe(true);
 
-    // log disarm to the dataflash
+    // log disarm
```
Comment (peterbarker) [inline ArduCopter/motors.cpp]: Missed two.

File: ArduSub/ArduSub.cpp [hunk 9]
```diff
@@ -336,7 +336,7 @@ void Sub::update_altitude()
     // read in baro altitude
     read_barometer();
 
-    // write altitude info to dataflash logs
+    // write altitude info to logger
```
Comment (peterbarker) [inline ArduSub/ArduSub.cpp]: Missed three

**Comment by peterbarker** [comment]
.... so feel free to just pop the last commit off.

**Comment by peterbarker** [comment]
I'm fixing this and will push it in once done.

**Comment by peterbarker** [comment]
Merged it.  Thanks Tom!

**Pull request closed without merging**